public-ui/src/app/components/db-no-results/no-results-search.component.tsx

```diff
@@ -0,0 +1,188 @@
+            loading: true
+        };
+    }
+
+    componentDidMount() {
+        this.fetchDomainTotals();
+    }
+
+    fetchDomainTotals() {
+        const {searchValue, measurementTestFilter, measurementOrderFilter} = this.props;
+        api.getDomainTotals(searchValue, measurementTestFilter, measurementOrderFilter).then(
+                result => {
+                    result.domainInfos = result.domainInfos.filter(domain =>
+                                        domain.standardConceptCount > 0);
+                    this.setState({domainInfoResults: result.domainInfos.filter(
+                    domain => domain.name.toLowerCase() !== 'physical measurements' &&
+                    domain.name.toLowerCase() !== 'fitbit'),
+                    surveyModuleResults: result.surveyModules,
+                    pmResults: result.domainInfos.filter(
+                    domain => domain.name.toLowerCase() === 'physical measurements'),
+                    fitbitResults: result.domainInfos.filter(
+                    domain => domain.name.toLowerCase() === 'fitbit')});
+                    this.setState({loading: false});
+                });
+    }
+
+    handleOnClick(domainInfo: any, type: string) {
+        const {searchValue, domainMatch} = this.props;
+        let url = '';
+        if (type === 'ehr') {
+            localStorage.setItem('ehrDomain', JSON.stringify(domainInfo));
+            url += 'ehr/' + domainToRoute[domainInfo.domain.toLowerCase()];
+            url += '?search=' + searchValue;
+            domainMatch();
+            navigateByUrl(url);
+        } else if (type === 'survey') {
+            localStorage.setItem('surveyModule', JSON.stringify(domainInfo));
+            url += 'survey/' + surveyIdToRoute[domainInfo.conceptId];
+            url += '?search=' + searchValue;
+            domainMatch();
+            navigateByUrl(url);
+        } else if (type === 'pm') {
+            url += 'physical-measurements/' + '/' + searchValue;
+            navigateByUrl(url);
+        } else if (type === 'fitbit') {
+            url += 'fitbit';
+            localStorage.setItem('searchText', searchValue);
+            navigateByUrl(url);
+        }
+    }
+
+    render() {
+        const {searchValue} = this.props;
+        const {loading, domainInfoResults, surveyModuleResults, pmResults, fitbitResults} = this.state;
+        return (
+            <React.Fragment>
+            <style>{styleCss}</style>
+            <div style={styles.noResults}>
+                { loading ?
+                <div style={styles.loadingDiv}>
+                    <p style={styles.loadingText}>Searching whole site for <strong>{searchValue} </strong></p>
+                    <div style={styles.spinnerDiv}><LoadingDots /></div>
+                </div>
+                : null
+                }
+                {
+                 domainInfoResults.map((domainInfo, index) => {
+                    const key = domainInfo.name + index;
```

Comment:
Nit: Not sure this is necessary since `index` is already unique. Any reason not to just use `index` as the key?